---
title: "Forest plot for age of ischemic stroke onset."
author: "Joanna von Berg; [Sander W. van der Laan, PhD](https://swvanderlaan.github.io) | @swvanderlaan | s.w.vanderlaan@gmail.com"
date: "`r Sys.Date()`"
output:
  html_notebook:
    cache: yes
    code_folding: hide
    collapse: yes
    df_print: paged
    fig.align: center
    fig_caption: yes
    fig_height: 6
    fig_retina: 2
    fig_width: 7
    highlight: tango
    theme: lumen
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
mainfont: Arial
subtitle: "A 'druggable-MI-targets' project"
editor_options:
  chunk_output_type: inline
---

```{r global_options, include = FALSE}
# further define some knitr-options.
knitr::opts_chunk$set(fig.width = 12, fig.height = 8, fig.path = 'Figures/', 
                      wwarning = TRUE, # show warnings during codebook generation
  message = TRUE, # show messages during codebook generation
  error = TRUE, # do not interrupt codebook generation in case of errors,
                # usually better for debugging
  echo = TRUE,  # show R code
                      eval = TRUE)

```

# Setup

We will clean the environment, setup the locations, define colors, and create a datestamp.

*Clean the environment.*

```{r echo = FALSE}

rm(list = ls())

```

*Set locations and working directories...*

```{r LocalSystem, echo = FALSE}
### Operating System Version 
### MacBook Pro
# ROOT_loc = "/Users/swvanderlaan/iCloud"
# AEROOT_loc = "/Users/swvanderlaan"
# STORAGE_loc = "/Volumes/LaCie/"
# STORAGE_loc = "/Users/swvanderlaan"

### MacBook Air
ROOT_loc = "/Users/slaan3/OneDrive - UMC Utrecht"
AEROOT_loc = "/Users/slaan3"
STORAGE_loc = "/Volumes/LaCie/"
# STORAGE_loc = "/Users/slaan3/"

GENOMIC_loc = paste0(ROOT_loc, "/Genomics")
AEDB_loc = paste0(AEROOT_loc, "/Athero-Express/AE-AAA_GS_DBs")
LAB_loc = paste0(GENOMIC_loc, "/LabBusiness")

PLINK_loc=paste0(STORAGE_loc,"/PLINK")
AEGSQC_loc =  paste0(PLINK_loc, "/_AE_ORIGINALS/AEGS_COMBINED_QC2018")
MICHIMP_loc=paste0(PLINK_loc,"/_AE_ORIGINALS/AEGS_COMBINED_EAGLE2_1000Gp3v5HRCr11")

GWAS_loc=paste0(PLINK_loc,"/_GWAS_Datasets/_SiGN")

PROJECT_loc = paste0(PLINK_loc, "/analyses/stroke/AAO_IschemicStroke")

# use this if there is relevant information here.
TARGET_loc = paste0(PROJECT_loc, "/targets")

### SOME VARIABLES WE NEED DOWN THE LINE
TRAIT_OF_INTEREST = "AAO" # Phenotype
PROJECTNAME = "AAO"

cat("\nCreate a new analysis directory...\n")
ifelse(!dir.exists(file.path(PROJECT_loc, "/",PROJECTNAME)), 
       dir.create(file.path(PROJECT_loc, "/",PROJECTNAME)), 
       FALSE)
ANALYSIS_loc = paste0(PROJECT_loc,"/",PROJECTNAME)

ifelse(!dir.exists(file.path(ANALYSIS_loc, "/PLOTS")), 
       dir.create(file.path(ANALYSIS_loc, "/PLOTS")), 
       FALSE)
PLOT_loc = paste0(ANALYSIS_loc,"/PLOTS")

ifelse(!dir.exists(file.path(PLOT_loc, "/QC")), 
       dir.create(file.path(PLOT_loc, "/QC")), 
       FALSE)
QC_loc = paste0(PLOT_loc,"/QC")

ifelse(!dir.exists(file.path(ANALYSIS_loc, "/OUTPUT")), 
       dir.create(file.path(ANALYSIS_loc, "/OUTPUT")), 
       FALSE)
OUT_loc = paste0(ANALYSIS_loc, "/OUTPUT")

ifelse(!dir.exists(file.path(ANALYSIS_loc, "/BASELINE")), 
       dir.create(file.path(ANALYSIS_loc, "/BASELINE")), 
       FALSE)
BASELINE_loc = paste0(ANALYSIS_loc, "/BASELINE")

setwd(paste0(PROJECT_loc))
getwd()
list.files()

```

*... a package-installation function ...*

```{r}
source(paste0(PROJECT_loc, "/scripts/functions.R"))

```

*... and load those packages.*

```{r loading_packages, message=FALSE, warning=FALSE}
source(paste0(PROJECT_loc, "/scripts/packages.R"))

```

```{r}
ggplot2::theme_set(ggplot2::theme_minimal())
pander::panderOptions("table.split.table", Inf)
```

_We will create a datestamp and define the Utrecht Science Park Colour Scheme_.
```{r Setting: Colors}

Today = format(as.Date(as.POSIXlt(Sys.time())), "%Y%m%d")
Today.Report = format(as.Date(as.POSIXlt(Sys.time())), "%A, %B %d, %Y")

### UtrechtScienceParkColoursScheme
###
### WebsitetoconvertHEXtoRGB:http://hex.colorrrs.com.
### Forsomefunctionsyoushoulddividethesenumbersby255.
###
###	No.	Color			      HEX	(RGB)						              CHR		  MAF/INFO
###---------------------------------------------------------------------------------------
###	1	  yellow			    #FBB820 (251,184,32)				      =>	1		or 1.0>INFO
###	2	  gold			      #F59D10 (245,157,16)				      =>	2		
###	3	  salmon			    #E55738 (229,87,56)				      =>	3		or 0.05<MAF<0.2 or 0.4<INFO<0.6
###	4	  darkpink		    #DB003F ((219,0,63)				      =>	4		
###	5	  lightpink		    #E35493 (227,84,147)				      =>	5		or 0.8<INFO<1.0
###	6	  pink			      #D5267B (213,38,123)				      =>	6		
###	7	  hardpink		    #CC0071 (204,0,113)				      =>	7		
###	8	  lightpurple	    #A8448A (168,68,138)				      =>	8		
###	9	  purple			    #9A3480 (154,52,128)				      =>	9		
###	10	lavendel		    #8D5B9A (141,91,154)				      =>	10		
###	11	bluepurple		  #705296 (112,82,150)				      =>	11		
###	12	purpleblue		  #686AA9 (104,106,169)			      =>	12		
###	13	lightpurpleblue	#6173AD (97,115,173/101,120,180)	=>	13		
###	14	seablue			    #4C81BF (76,129,191)				      =>	14		
###	15	skyblue			    #2F8BC9 (47,139,201)				      =>	15		
###	16	azurblue		    #1290D9 (18,144,217)				      =>	16		or 0.01<MAF<0.05 or 0.2<INFO<0.4
###	17	lightazurblue	  #1396D8 (19,150,216)				      =>	17		
###	18	greenblue		    #15A6C1 (21,166,193)				      =>	18		
###	19	seaweedgreen	  #5EB17F (94,177,127)				      =>	19		
###	20	yellowgreen		  #86B833 (134,184,51)				      =>	20		
###	21	lightmossgreen	#C5D220 (197,210,32)				      =>	21		
###	22	mossgreen		    #9FC228 (159,194,40)				      =>	22		or MAF>0.20 or 0.6<INFO<0.8
###	23	lightgreen	  	#78B113 (120,177,19)				      =>	23/X
###	24	green			      #49A01D (73,160,29)				      =>	24/Y
###	25	grey			      #595A5C (89,90,92)				        =>	25/XY	or MAF<0.01 or 0.0<INFO<0.2
###	26	lightgrey		    #A2A3A4	(162,163,164)			      =>	26/MT
### 
###	ADDITIONAL COLORS
###	27	midgrey			#D7D8D7
###	28	verylightgrey	#ECECEC"
###	29	white			#FFFFFF
###	30	black			#000000
###----------------------------------------------------------------------------------------------

uithof_color = c("#FBB820","#F59D10","#E55738","#DB003F","#E35493","#D5267B",
                 "#CC0071","#A8448A","#9A3480","#8D5B9A","#705296","#686AA9",
                 "#6173AD","#4C81BF","#2F8BC9","#1290D9","#1396D8","#15A6C1",
                 "#5EB17F","#86B833","#C5D220","#9FC228","#78B113","#49A01D",
                 "#595A5C","#A2A3A4", "#D7D8D7", "#ECECEC", "#FFFFFF", "#000000")

uithof_color_legend = c("#FBB820", "#F59D10", "#E55738", "#DB003F", "#E35493",
                        "#D5267B", "#CC0071", "#A8448A", "#9A3480", "#8D5B9A",
                        "#705296", "#686AA9", "#6173AD", "#4C81BF", "#2F8BC9",
                        "#1290D9", "#1396D8", "#15A6C1", "#5EB17F", "#86B833",
                        "#C5D220", "#9FC228", "#78B113", "#49A01D", "#595A5C",
                        "#A2A3A4", "#D7D8D7", "#ECECEC", "#FFFFFF", "#000000")
### ----------------------------------------------------------------------------
```

# Introduction

We will create a forest plot of the association of the _ApoE_ variant with age of onset of stroke in all groups. 

# Setting the NPG colors

```{r}
library("scales")
pal_npg("nrc")(10)
show_col(pal_npg("nrc")(10))

# show_col(pal_npg("nrc", alpha = 0.6)(10))

```

# Prepare data

First we prepare the data for the forest plot. 
```{r}
snps <- read_table(file = "SNP Chr BP GenPos Allele1 Allele0 A1Freq Info Beta SE P A0Freq
rs429358 19 45411941 0 T C 0.865926 0.961725 1.21275 0.252633 1.6e-6 0.13407400000000003
rs429358 19 45411941 0 T C 0.868345 0.961725 1.77608 0.401491 9.7e-6 0.13165499999999997
rs429358 19 45411941 0 T C 0.864025 0.961725 0.802006 0.321942 0.013 0.13597499999999996",
col_names = TRUE)
snps

snps$analysis <- factor(c("all", "women", "men"), ordered = TRUE)
snps %>%
  mutate(ci.l = Beta - 1.96*SE) %>%
  mutate(ci.h = Beta + 1.96*SE) %>%
  mutate(locus = "ApoE") -> snps

```



```{r}
adj <- 0.33333

# pdf(file = paste0(PLOT_loc, "/21.04.30.forest.SiGN.pdf"), width = 18, height = 10)
  nsnps <- 1
  nph <- 3
  
  # set the colors
  #cols <- rgb(c(252,209,145,62,25), c(255,219,170,96,52), c(245,189,157,111,65), maxColorValue = 255)
  # cols <- rgb(c(20,250,243,254,99), c(117,126,236,203,204), c(135,92,229,95,200), maxColorValue = 255)
  cols <- c("black", uithof_color[25], "#DC0000FF", "#00A087FF", "#4DBBD5FF")
  
  snps %>%
    mutate(Beta = -Beta) %>%
    ggplot(aes(x = Beta, xmax = -ci.l, xmin = -ci.h, 
               y = locus, 
               fill = analysis, shape = analysis, colour = analysis, label = analysis)) +
    theme_minimal() +
    theme(panel.grid.major.y = element_line(size = 0), 
          panel.grid.minor.y = element_line(size = 0), 
          panel.grid.minor = element_line(colour = cols[2], size = 0.2, linetype = 2), 
          panel.grid.major = element_line(colour = cols[2], size = 0.4, linetype = 2), 
          axis.ticks = element_line(colour = cols[1]),
          text = element_text(size = 15, color = cols[1]), 
      axis.title.x = element_text(vjust = -1, color = cols[1]), 
      plot.margin = unit(x = c(0,0,1,0), units = "cm"), 
      axis.text.x = element_text(color = cols[1]), 
      axis.text.y = element_text(size = 0)) +
    # labs(title = "Effect of aao SNPs in MEGASTROKE") +
    geom_vline(xintercept = 0, linetype = 1, colour = cols[1], size = 1) +
    xlim(c(-3,1)) +
    ylab(label = "") +
    xlab(label = "effect of rs429358 on age at onset of ischemic stroke [in years]") +
    scale_shape_manual(values = 15:19) +
    geom_point(size = 6, position = position_nudge(y = c(0, adj, -adj))) +
    scale_colour_manual(values = cols[c(3,4,5)]) +
    geom_errorbarh(size = 2, height = 0.05, position = position_nudge(y = c(0, adj, -adj))) +
    geom_label(size = 5, color = cols[2], fill = NA, label.size = 0, position = position_nudge( y = c(0, adj, -adj)+0.15)) +
    guides(shape = FALSE, colour = FALSE, label = FALSE, fill = FALSE)

# dev.off()

ggsave(file = paste0(PLOT_loc, "/21.04.30.forest.SiGN.eps"), last_plot())
ggsave(file = paste0(PLOT_loc, "/21.04.30.forest.SiGN.pdf"), last_plot())
# ggsave(file = paste0(PLOT_loc, "/21.04.30.forest.SiGN.png"), last_plot())

```


# Session information

------

    Version:      v1.0.0
    Last update:  2022-05-06
    Written by:   Sander W. van der Laan (s.w.vanderlaan-2[at]umcutrecht.nl).
    Description:  Script to create forest plot.
    Minimum requirements: R version 3.4.3 (2017-06-30) -- 'Single Candle', Mac OS X El Capitan
    
    Changes log
    * v1.0.0 Initial version. 

------

```{r eval = TRUE}
sessionInfo()
```


# Saving environment
```{r Saving}

save.image(paste0(PROJECT_loc, "/",Today,".",PROJECTNAME,".ForestPlot.RData"))
```


------
<sup>&copy; 1979-2022 Sander W. van der Laan | s.w.vanderlaan[at]gmail.com | [swvanderlaan.github.io](https://swvanderlaan.github.io).</sup>
------


